signature verification is failing

**Describe the bug**

Signature verification is failing for `v1.6.0-beta5`.

**Steps to reproduce**

```
❯ asdf install opentofu latest
redirect url: https://github.com/opentofu/opentofu/releases/tag/v1.6.0-beta5
* Downloading opentofu release v1.6.0-beta5...
Verifying signatures and checksums
* Downloading signature files ...
Error: none of the expected identities matched what was in the certificate, got subjects [https://github.com/opentofu/opentofu/.github/workflows/release.yml@refs/heads/main] with issuer https://token.actions.githubusercontent.com
main.go:74: error during command execution: none of the expected identities matched what was in the certificate, got subjects [https://github.com/opentofu/opentofu/.github/workflows/release.yml@refs/heads/main] with issuer https://token.actions.githubusercontent.com
signature verification failed
```

beta4 work

```
$ wget https://github.com/opentofu/opentofu/releases/download/v1.6.0-beta4/tofu_1.6.0-beta4_SHA256SUMS
$ wget https://github.com/opentofu/opentofu/releases/download/v1.6.0-beta4/tofu_1.6.0-beta4_SHA256SUMS.pem
$ wget https://github.com/opentofu/opentofu/releases/download/v1.6.0-beta4/tofu_1.6.0-beta4_SHA256SUMS.sig
$ cosign verify-blob --signature tofu_1.6.0-beta4_SHA256SUMS.sig --certificate tofu_1.6.0-beta4_SHA256SUMS.pem --certificate-identity https://github.com/opentofu/opentofu/.github/workflows/release.yml@refs/tags/v1.6.0-beta4 --certificate-oidc-issuer=https://token.actions.githubusercontent.com tofu_1.6.0-beta4_SHA256SUMS
Verified OK
```

beta5 fails
```
$ wget https://github.com/opentofu/opentofu/releases/download/v1.6.0-beta5/tofu_1.6.0-beta5_SHA256SUMS
$ wget https://github.com/opentofu/opentofu/releases/download/v1.6.0-beta5/tofu_1.6.0-beta5_SHA256SUMS.sig
$ wget https://github.com/opentofu/opentofu/releases/download/v1.6.0-beta5/tofu_1.6.0-beta5_SHA256SUMS.pem
$ cosign verify-blob --signature tofu_1.6.0-beta5_SHA256SUMS.sig --certificate tofu_1.6.0-beta5_SHA256SUMS.pem --certificate-identity https://github.com/opentofu/opentofu/.github/workflows/release.yml@refs/tags/v1.6.0-beta5 --certificate-oidc-issuer=https://token.actions.githubusercontent.com tofu_1.6.0-beta5_SHA256SUMS
Error: none of the expected identities matched what was in the certificate, got subjects [https://github.com/opentofu/opentofu/.github/workflows/release.yml@refs/heads/main] with issuer https://token.actions.githubusercontent.com
main.go:74: error during command execution: none of the expected identities matched what was in the certificate, got subjects [https://github.com/opentofu/opentofu/.github/workflows/release.yml@refs/heads/main] with issuer https://token.actions.githubusercontent.com
```

--- a/lib/utils.bash
+++ b/lib/utils.bash
@@ -117,7 +117,7 @@ verify() {
 	local -r checksum_file="${TOOL_BIN_NAME}_${version}_SHA256SUMS"
 	local -r signature_file="${checksum_file}.sig"
 	local -r cert_file="${checksum_file}.pem"
-	local -r cert_identity="https://github.com/opentofu/opentofu/.github/workflows/release.yml@refs/tags/v${version}"
+	local -r cert_identity="https://github.com/opentofu/opentofu/.github/workflows/release.yml@refs/head/main"
 	local -r cert_oidc_issuer="https://token.actions.githubusercontent.com"
 
 	baseURL="$GH_REPO/releases/download/v${version}"

